api/processors/group.py

```diff
@@ -52,18 +59,55 @@ def _create_log_events_start(cls, group: Group) -> None:
             for teacher in group.teachers.iterator()
         )
 
+    @classmethod
+    def _create_log_events_abort(cls, group: Group) -> None:
+        GroupLogEvent.objects.create(group=group, type=GroupLogEventType.ABORTED)
+
+        CoordinatorLogEvent.objects.bulk_create(
+            CoordinatorLogEvent(
+                coordinator=coordinator,
+                group=group,
+                type=CoordinatorLogEventType.TOOK_NEW_GROUP,
+            )
+            for coordinator in group.coordinators.iterator()
+        )
+
+        StudentLogEvent.objects.bulk_create(
+            StudentLogEvent(
+                student=student, from_group=group, type=StudentLogEventType.GROUP_ABORTED
+            )
+            for student in group.students.iterator()
+        )
+
+        TeacherLogEvent.objects.bulk_create(
+            TeacherLogEvent(
+                teacher=teacher, from_group=group, type=TeacherLogEventType.GROUP_ABORTED
+            )
+            for teacher in group.teachers.iterator()
+        )
+
     @classmethod
     def _set_statuses_start(cls, group: Group) -> None:
         timestamp = timezone.now()
 
         cls._set_status(obj=group, status=GroupStatus.WORKING, status_since=timestamp)
 
-        cls._set_coordinators_status_start(timestamp)
+        cls._set_coordinators_status_start_or_abort(timestamp)
         cls._set_students_status_start(group=group, timestamp=timestamp)
         cls._set_teachers_status_start(timestamp)
 
+    @classmethod
+    def _set_statuses_abort(cls, group: Group) -> None:
+        timestamp = timezone.now()
+
+        cls._set_status(obj=group, status=GroupStatus.ABORTED, status_since=timestamp)
+
+        cls._set_students_status_abort(group=group, timestamp=timestamp)
+        cls._set_teachers_status_abort(timestamp)
+        cls._set_coordinators_status_start_or_abort(timestamp)
```

Comment:
I know it's might seem premature, but it's clear now already that at least 4 group events will trigger same status changes for coordinators: start, transfer, abort and finish.

So we can either call this function `cls._set_coordinators_status_start_transfer_abort_finish()` or (I find this option more flexible and readable) factor out this functionality into a method called something like `cls._set_coordinators_status_common()` and then have separate methods `cls._set_coordinators_status_start()`, `cls._set_coordinators_status_abort()` etc. simply call this method. This way the code will be readable and we'll have more flexibility as to what happens after each of the events.

Reply:
OK nevermind :) let's keep it this way for now